backend/response_codes.js

```diff
@@ -485,6 +486,13 @@
 		// next();
 	};
 
+	responseCodes.writeHead =  function (place, req, res, next, customHeaders) {
```

Comment:
this is confusing, we have a function to write a standard response, and we have a function to write a streamed response.

You're essentially replying with a standard response, without any body, and some extra customheaders. don't write a new function for this, just expand the existing function to take in custom headers.